app/services/legacy_sync_service.rb

```diff
@@ -0,0 +1,31 @@
+class LegacySyncService
```

Comment:
this class may better be expressed as a module... it's got one instance var and otherwise just runs methods of other objects

Reply:
I thought the reason for having a class over a module was because class stores state, and module doesn't. But we are initialising with a dataset, so does that not suggest class is better? Am just asking for learning purposes!

Reply:
My argument is that the state doesn't need to be stored, and everywhere this class is being used it's being called like a function (initialise followed by immediate method call)

Reply:
I agree, it doesn't need to be stored, but it's nice to :)

Reply:
I prefer to use a module as a "bag of methods" that I can include into multiple classes to extend them.

In this case, class or module is just a matter of preference.